### PR TITLE
Rename Azure common to Azure core.

### PR DESCRIPTION
**Pull request by sima-zhu**
Issue #3393

### CODE REVIEW

File: pom.client.xml [hunk 1]
```diff
@@ -115,14 +115,14 @@
     <groovy-eclipse-batch.version>2.4.16-03</groovy-eclipse-batch.version>
     <client-runtime.version.v2>2.1.1</client-runtime.version.v2>
     <slf4j-api.version>1.7.5</slf4j-api.version>
-    <azure-common.version>1.0.0-SNAPSHOT</azure-common.version>
-    <azure-common-auth.version>1.0.0-SNAPSHOT</azure-common-auth.version>
-    <azure-common-mgmt.version>1.0.0-SNAPSHOT</azure-common-mgmt.version>
+    <azure-core.version>1.0.0-SNAPSHOT</azure-core.version>
+    <azure-core-auth.version>1.0.0-SNAPSHOT</azure-core-auth.version>
+    <azure-core-mgmt.version>1.0.0-SNAPSHOT</azure-core-mgmt.version>
```
Comment (conniey) [inline pom.client.xml]: I thought the rename to management was also part of this PR? Or is that in another?

File: core/azure-core-mgmt/src/main/java/com/azure/core/mgmt/AsyncOperationResource.java [hunk 2]
```diff
@@ -1,7 +1,7 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
 
-package com.azure.common.mgmt;
+package com.azure.core.mgmt;
```
Comment (JonathanGiles) [inline core/azure-core-mgmt/src/main/java/com/azure/core/mgmt/AsyncOperationResource.java]: As @conniey has stated, and as I discussed in the standup today, `mgmt` should be `management`, as outlined in #3349, and it should be done as part of this PR ideally (to lessen the pain to one PR).

Comment (sima-zhu) [reply core/azure-core-mgmt/src/main/java/com/azure/core/mgmt/AsyncOperationResource.java]: Done with the renaming mgmt. to management for azure core.

File: pom.client.xml (hunk 1 above)
Comment (sima-zhu) [reply pom.client.xml]: Will do in next commit. Thanks for reminding!

**Comment by sima-zhu** [comment]
The mgmt renaming is done with the newest commit.

**Pull request merged**